### PR TITLE
chore: drop Node.js 12

### PR DESCRIPTION
**Pull request by aqrln**
Ref: https://github.com/prisma/client-planning/issues/19

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -402,11 +402,9 @@ jobs:
       fail-fast: false
       matrix:
         node:
-          - 12
           - 14
           - 15
           - 16
-          - 12.6.0 # minimal node version via https://github.com/prisma/prisma/blob/master/src/packages/client/package.json and minimal minor version of node 12 via https://www.prisma.io/docs/reference/system-requirements
           - 14.0.0 # minimal minor version of node 14 via https://www.prisma.io/docs/reference/system-requirements
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
          - 14.0.0 # minimal node version via https://github.com/prisma/prisma/blob/master/src/packages/client/package.json and minimal minor+patch version of node 14 via https://www.prisma.io/docs/reference/system-requirements
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -402,11 +402,9 @@ jobs:
       fail-fast: false
       matrix:
         node:
-          - 12
           - 14
           - 15
           - 16
```
Comment (janpio) [inline .github/workflows/test.yaml]:
```suggestion
          - 14 # latest minor+patch oh GH Actions
          - 15 # latest minor+patch oh GH Actions
          - 16 # latest minor+patch oh GH Actions
```

**Comment by janpio** [comment]
There might be some cloud platforms where the app is configured to run Node 12 - we'll see when we actually merged the change in the min requirement.

**Review by Jolg42 — APPROVED** [review]
I fixed gcp and azure functions to node 14

**Comment by Jolg42** [comment]
Also I logged into the EC2 graviton ARM to set 14 as default and set Vercel ones to 14.x as they were already faiing on dev branch

**Comment by Jolg42** [comment]
I changed netlify settings:
https://app.netlify.com/sites/prisma2-e2e-tests-netlify-ci/settings/deploys
NODE_VERSION from 12 to 14

Note that for https://app.netlify.com/sites/prisma2-e2e-tests-netlify-cli/settings/deploys I found this, it looks very outdated 
ZISI_VERSION = 0.4.0-8
since latest version is [v5.10.0](https://github.com/netlify/zip-it-and-ship-it/releases/tag/v5.10.0)

**Comment by Jolg42** [comment]
@aqrln looks ready to merge, the errors are flaky and @millsp is working on the dataproxy errors

**Comment by aqrln** [comment]
Thank you @Jolg42!

**Pull request merged**